### PR TITLE
More Prop simplifications

### PR DESCRIPTION
**Pull request by msooseth**
## Description

This does a number of Prop simplifications. I needed it to debug another issue. Basically:

* `[PAnd (PAnd x y) z]` now becomes `[x, y, z]`, previously it was `[x, PAnd y z]`
* Double negation not only works for `Eq` inside Prop, but also for `LT`, `GT`, etc

This makes it a lot easier to debug some issues (much less deep trees), and accidentally makes things faster, too. Was not the intention, though.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [x] updated the changelog

### CODE REVIEW

File: src/EVM/Expr.hs [hunk 1]
```diff
@@ -1292,8 +1298,13 @@ simplifyProp prop =
...
 flattenProps [] = []
 flattenProps (a:ax) = case a of
-  PAnd x1 x2 -> x1:x2:flattenProps ax
+  PAnd x1 x2 -> (subFlatten x1) ++ (subFlatten x2) ++ flattenProps ax
   x -> x:flattenProps ax
+  where
+    subFlatten :: Prop -> [Prop]
+    subFlatten b = case b of
+      PAnd y1 y2 -> flattenProps [y1,y2]
+      _ -> [b]
```
Comment (blishko) [inline src/EVM/Expr.hs:1307]: Can we do without the helper function? Like this?
```haskell
flattenProps :: [Prop] -> [Prop]
flattenProps [] = []
flattenProps (a:ax) = case a of
  PAnd x1 x2 -> flattenProps [x1] ++ flattenProps [x2] ++ flattenProps ax
  x -> x:flattenProps ax
```

Would the helper function be more efficient for any reason?

File: src/EVM/Expr.hs [hunk 2]
```diff
@@ -1224,7 +1224,6 @@ simplifyProp prop =
     go (PNeg (PNeg a)) = a
 
     -- solc specific stuff
-    go (PEq (IsZero (Eq a b)) (Lit 0)) = PEq a b
```
Comment (blishko) [inline src/EVM/Expr.hs:1227]: Why was this removed?

**Review by blishko — COMMENTED** [review]
Looks good, I just have some questions.

File: src/EVM/Expr.hs (hunk 2 above)
Comment (msooseth) [reply src/EVM/Expr.hs:1227]: It was only moved lower, to the block about double negation, I think it fits there better?

File: src/EVM/Expr.hs (hunk 1 above)
Comment (msooseth) [reply src/EVM/Expr.hs]: Oh, of course! Good point :D Fixed!
